src/core/ui/egl/egl-ui.cpp

```diff
@@ -0,0 +1,93 @@
+#if OTTO_UI_EGL
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <chrono>
+#include <cmath>
+#include <cstdio>
+#include <string>
+#include <thread>
+
+#include "core/globals.hpp"
+#include "core/ui/mainui.hpp"
+
+#define NANOVG_GLES2_IMPLEMENTATION
```

Comment:
It can't be in the "deps" file -- that macro causes nanovg to actually instantiate some things.  If it's placed somewhere that multiple files "see" it, you get name collisions at link time.

Reply:
oh, interesting - we should maybe look at only including nanovg_gl.h here then, i also like avoiding too many C apis included in headers because of namespace clutter (and macros)